### PR TITLE
feat: improve ui of Opsgenie

### PR DESCRIPTION
**Pull request by sharad-sharma**

### CODE REVIEW

**Review by niveshm — COMMENTED** [review]
Fix tests for validate request

**Comment by sharad-sharma** [comment]
> Fix tests for validate request

Doesn't have tests for validate request of Opsgenie.

File: opsgenie_access/access.py [hunk 1]
```diff
@@ -40,16 +40,16 @@ def validate_request(self, access_labels_data, request_user, is_group=False):
         total_teams_list = helper.teams_list()
         if total_teams_list is None:
             raise Exception(constants.TEAM_LIST_ERROR)
-        for access_label_data in access_labels_data[0]["teams_list"]:
+        for access_label_data in json.loads(access_labels_data.get("selected-opsgenie-teams")):
```
Comment (niveshm) [inline opsgenie_access/access.py:43]:
```suggestion
        for access_label_data in json.loads(access_request_form.get("selected-opsgenie-teams")):
```

File: opsgenie_access/templates/opsgenie_access/access_request_form.html [hunk 2]
```diff
@@ -1,96 +1,231 @@
...
-      <div class="col-sm-3">
-        <label for="Access" style="font-weight: bolder; min-height:0px; margin-top:10px; font-size:medium;">Access Level:</label>
-        <select id="opsgenieAccessLevel" name="opsgenieAccessLevel" class="form-control" required>
-          <option hidden disabled selected value> -- select -- </option>
-          <option value="user" >Member</option>
-          <option value="team_admin" >Admin</option>
-        </select>
-      </div>
-    </div>
+  const selectedList = {}
```
Comment (niveshm) [inline opsgenie_access/templates/opsgenie_access/access_request_form.html:3]: change this variable something specific to opsgenie
```suggestion
  const opsgenieSelectedList = {}
```

File: opsgenie_access/templates/opsgenie_access/access_request_form.html [hunk 3]
```diff
@@ -1,96 +1,231 @@
...
+
+    for (iter = 0; iter < teams.length; iter++) {
+      if(!selectedList[$(teams[iter]).attr('team_name')]) {
+        addOpsgenieTeamSelection(teams[iter]);
       }
-      labelComponents[0]["teams_list"] = getSelectedTeams()
-      labelComponents[0]["UserType"] = getOpsgenieAccessLevel()
-      $('#opsgenieLabel').val(JSON.stringify(labelComponents));
     }
+    updateSelectedOpsgenieTeam();
```
Comment (niveshm) [inline opsgenie_access/templates/opsgenie_access/access_request_form.html:83]: We dont need this here right, we are already doing it in line 92

File: opsgenie_access/templates/opsgenie_access/access_request_form.html (hunk 2 above)
Comment (sharad-sharma) [reply opsgenie_access/templates/opsgenie_access/access_request_form.html]: done.

**Review by niveshm — APPROVED** [review]

**Review by kneeraj2000 — APPROVED** [review]

**Pull request merged**